R/page.R

```diff
@@ -397,31 +397,44 @@ page_navbar <- function(
   fillable_mobile = FALSE,
   gap = NULL,
   padding = NULL,
-  position = c("static-top", "fixed-top", "fixed-bottom"),
   header = NULL,
   footer = NULL,
-  bg = NULL,
-  inverse = "auto",
-  underline = TRUE,
-  collapsible = TRUE,
+  navbar_options = NULL,
   fluid = TRUE,
   theme = bs_theme(),
   window_title = NA,
-  lang = NULL
+  lang = NULL,
+  position = deprecated(),
+  bg = deprecated(),
+  inverse = deprecated(),
+  underline = deprecated(),
+  collapsible = deprecated()
 ) {
 
   sidebar <- maybe_page_sidebar(sidebar)
 
   padding <- validateCssPadding(padding)
   gap <- validateCssUnit(gap)
 
+  # Change behavior when called by Shiny
+  # TODO: Coordinate with next bslib version bump in Shiny to use the new interface
+  was_called_by_shiny <- isNamespaceLoaded("shiny") && identical(rlang::caller_fn(), shiny::navbarPage)
+
+  .navbar_options <- navbar_options_resolve_deprecated(
+    options_user = navbar_options,
+    position = position,
+    bg = bg,
+    inverse = inverse,
+    collapsible = collapsible,
+    underline = underline,
+    .fn_caller = "page_navbar",
+    .warn_deprecated = !was_called_by_shiny
```

Comment:
I would really really like to decouple `shiny::navbarPage()` from `bslib::page_navbar()`. I'm thinking it'd be an update to add `page_navbar_legacy()` that we'd export but mark as internal so it doesn't show up in docs and indices.

What I want out of that change is to have a clear line around what parts of the navbar API we shouldn't touch without thinking carefully about backwards compatibility or API changes and which parts are owned by bslib. Currently it feels too entangled and over time entropy wins leading to more entanglement.